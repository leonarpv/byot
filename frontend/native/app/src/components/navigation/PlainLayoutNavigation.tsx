import * as React from 'react';
import {Screens} from '../../navigation/Screens';
import {DefaultNavigatorOptions} from '@react-navigation/native';
import {FullScreenKeyboardAvoiding} from '../screens/FullScreenKeyboardAvoiding';
import {PlainLayoutNarrow} from '../plain-layout/PlainLayoutNarrow';
import {Colors} from 'react-native-ui-lib';
import {StyleSheet} from 'react-native';
import {StackNavigationOptions} from '@react-navigation/stack/lib/typescript/src/types';
import {
  PlainLayoutBackButtonPortalIn,
  PlainLayoutBackButtonPortalProvider,
} from './PlainLayoutBackButtonPortal';

interface Props {
  children: React.ReactNode;
  navigatorComponent: React.ComponentType<DefaultNavigatorOptions<StackNavigationOptions>>;
}

interface State {}

const makeStyles = (props: Props, state: State) =>
  StyleSheet.create({
    card: {
      backgroundColor: Colors.white,
      flex: 1,
    },
  });

export const PlainLayoutNavigation: React.FC<Props> = (props: Props) => {
  const styles = makeStyles(props, {});
  const Navigator = props.navigatorComponent;
  return (
    <FullScreenKeyboardAvoiding>
      <PlainLayoutBackButtonPortalProvider>
        <PlainLayoutNarrow outside={<PlainLayoutBackButtonPortalIn />}>
          <Navigator
            initialRouteName={Screens.Login.Name}
            screenOptions={{
              headerShown: false,
              cardOverlayEnabled: false,
              cardShadowEnabled: false,
              cardStyle: styles.card,
            }}>
            {props.children}
          </Navigator>
        </PlainLayoutNarrow>
      </PlainLayoutBackButtonPortalProvider>
    </FullScreenKeyboardAvoiding>
  );
};
