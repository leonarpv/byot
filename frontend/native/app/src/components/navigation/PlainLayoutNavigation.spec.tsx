import * as React from 'react';
import {render} from '@testing-library/react-native';
import {PlainLayoutNavigation} from './PlainLayoutNavigation';
import {Text, View} from 'react-native';

describe('<PlainLayoutNavigation/>', () => {
  it('should render', () => {
    const {asJSON} = render(
      <PlainLayoutNavigation navigatorComponent={View as any}>
        <Text>Text</Text>
      </PlainLayoutNavigation>
    );
    expect(asJSON()).toMatchSnapshot();
  });
});
