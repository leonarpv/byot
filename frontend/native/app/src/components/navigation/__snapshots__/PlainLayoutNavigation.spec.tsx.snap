// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainLayoutNavigation/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#FAFAF1",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
            "paddingTop": 20,
          },
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../native/app/src/assets/byoT-full.png",
          }
        }
        style={
          Array [
            undefined,
            Object {
              "height": 100,
              "resizeMode": "contain",
              "width": 140.41666666666666,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <View
          initialRouteName="Login"
          screenOptions={
            Object {
              "cardOverlayEnabled": false,
              "cardShadowEnabled": false,
              "cardStyle": Object {
                "backgroundColor": "#FAFAF1",
                "flex": 1,
              },
              "headerShown": false,
            }
          }
        >
          <Text>
            Text
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
